x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/service_accounts/10_basic.yml

```diff
@@ -85,7 +85,7 @@ teardown:
 
   - match: { "error.type": "security_exception" }
   - match:
-      error.reason: "action [cluster:admin/xpack/security/user/delete] is unauthorized for user [elastic/fleet-server], this action is granted by the cluster privileges [manage_security,all]"
+      error.reason: "/action.\\[cluster:admin/xpack/security/user/delete\\].is.unauthorized.for.*\\[elastic/fleet-server\\].*this.action.is.granted.by.the.cluster.privileges.*/"
```

Comment:
I don't understand why the regex needs a `.` in place of a space. Can you explain?

Reply:
The match regex assertion is compiled with the `COMMENTS` option that ignores whitespaces and permits comments (`#`). 

https://github.com/elastic/elasticsearch/blob/54d69182fdf316375aa3f5df6d1badf970a10851/test/framework/src/main/java/org/elasticsearch/test/rest/yaml/section/MatchAssertion.java#L67

We can also assert whitespace with `\ `, e.g. `this\ action\ is\ granted\ by\ the\ cluster\ privileges`. But I think escapes looks ugly and it's not all that important. Many existing yaml tests just use `.`.